nimbella/nimbella.py

```diff
@@ -49,13 +48,14 @@ def storage(web=False):
         raise Exception('Object store credentials are not available.')
     try:
         creds = json.loads(creds)
-        client = gstorage.Client(
-            creds['project_id'], credentials=service_account.Credentials.from_service_account_info(creds))
     except:
         raise Exception(
             'Insufficient information in provided credentials or credentials were invalid.')
 
-    hostpart = '-'.join(apiHost.replace('https://', '').split('.'))
-    datapart = '' if web else 'data-'
-    bucket = datapart + namespace + '-' + hostpart
-    return client.get_bucket(bucket)
+    provider_id = creds.get('provider', '@nimbella/storage-gcs')
+    plugin = plugin_manager.find_plugin(provider_id)
+    # TODO: Cannot find provider?
```

Comment:
I'm not too worried about this assuming the actually occurring case of provider=None is handled.  It would mean that the field is present but corrupted.   But, perhaps there is _something_ we could do to indicate the nature of the problem for serviceability?

Reply:
Thinking about this.... we should definitely log an error message to indicate the problem. Will add something.

Reply:
I've added an exception to be thrown in this (very unlikely) case.